### ISSUE TITLE
Cunning locking code can deadlock an answer

### ISSUE THREAD

**Issue by GoogleCodeExporter** [dlsgusrn7577/mockito#190]
```
* What steps will reproduce the problem?

Please find attached test class to reproduce the problem.

* What is the expected output? What do you see instead?

Mocks cannot be shared across the threads. When I try to call the same
method from one mock instance in different threads, it is executed in
series, instead of in parallel. It seems that something is synchronized in
Mockito.

* What version of the product are you using? On what operating system?

Mockito 1.8.4, Windows XP
```

Original issue reported on code.google.com by `mszczyto...@gmail.com` on 20 May 2010 at 1:00

Attachments:
- [MockitoTest.java](https://storage.googleapis.com/google-code-attachments/mockito/issue-190/comment-0/MockitoTest.java)

**Comment by GoogleCodeExporter** [comment]
Original comment by `szcze...@gmail.com` on 1 Jun 2010 at 8:50
- Changed title: **Cunning locking code can deadlock an answer**

**Comment by GoogleCodeExporter** [comment]
```
This issue was closed by revision r2013.
```

Original comment by `szcze...@gmail.com` on 1 Jun 2010 at 8:51
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
Original comment by `szcze...@gmail.com` on 3 Jul 2011 at 12:43
- Added labels: **Milestone-Release1.9-rc1**

**Comment by GoogleCodeExporter** [comment]
```
Issue 265 has been merged into this issue.
```

Original comment by `brice.du...@gmail.com` on 3 Oct 2011 at 9:52
